book/src/troubleshooting.md

```diff
@@ -0,0 +1,91 @@
+<!-- toc -->
+
+
+## What We Would Usually Do
+
+Regardless of the issue, these are the "default" troubleshooting steps that are good to keep in mind:
+
+1. diagnose: [run `cargo dist plan`](./way-too-quickstart.md#check-what-ci-will-build)
+2. update and repair: [run `cargo dist init` again](./updating.md)
+3. test your process: [try `pr-run-mode = "upload"`](./ci/github.md#build-and-upload-artifacts-on-every-pull-request)
+
+These are also great steps to follow proactively, if you're updating your cargo-dist config.
+
+
+
+## Nothing To Release / Missing Packages / Too Many Packages
+
+cargo-dist tries to support as many release workflows as possible, and that means it needs you to tell it what you're interested in releasing. There are several ways to opt things in and out of being released, the most important are:
+
+* [git tag formats](./workspaces/workspace-guide.md#announcement-tags)
+    * git tags select which packages you're interested in doing a release for
+    * do all the packages you want to publish have the same version as your git tag?
+    * are you prefixing the tag with something that looks like a package name?
+* [`[package].publish`](./reference/config.md#publish)
+    * tells cargo whether the package should be published to crates.io
+    * cargo-dist assumes you don't want to release `publish = false` packages (since they're probably for testing)
+* [`[package.metadata.dist].dist`](./reference/config.md#dist)
+    * overrides `publish` for cargo-dist releases, either to force a package on or off
+
+
+
+## Recovering Failed CI Jobs
+
+Sometimes CI fails, and that's ok! The steps to follow depend on what went wrong. There's little that can't be recovered, you've got this.
+
+
+### Failed CI: Spurious
+
+If you believe the failure was spurious (Github CI flaking out, some networked service being temporarily down, a SECRET not being set...), then good news: it's totally safe to "retry failed jobs" in the Github CI interface! We should pick up your release process from where it left off.
+
+We *DO NOT* recommend "retry all jobs". Either it's redundant or it can cause problems with trying to repeat side-effects like publishing a package.
+
+
+### Failed CI: Busted Builds
+
+If you believe something was busted in your release process, and the commit you tried to release from isn't suitable, that's generally ok! Usually this will occur because some part of your build is broken, perhaps only when releasing with cargo-dist, or only on a particular platform.
+
+If this is the case then presumably your release process errored out before the the "host" step where we actually uploaded anything, so good news: no side effects need to be rolled back!
+
+Well, one side-effect needs to be dealt with, but it's the one you did to kick off the release: [delete the git tag from github and your local machine](https://stackoverflow.com/questions/5480258/how-can-i-delete-a-remote-tag), get your build sorted out, and then tag the new release commit.
```

Comment:
We should avoid telling people to delete and retag, IMO.

Reply:
I don't love it either but people get very precious about version numbers, so if we don't explain it they might mess things up worse.

Reply:
Yeah, that's fair.